gamebook/scene.py

```diff
@@ -57,10 +57,13 @@ def next_scene_draw(self, option_user_key):
         if not option:
             raise Exception("this option does not exist in this scene")
 
-        if type(option.next_scenes) == list:
-            weights = [i.get_success_rate() for i in option.next_scenes]
+        if option.next_scenes:
+            if type(option.next_scenes) == list:
```

Comment:
I'd swap type == list to check if it is not, and remove the else, like so:
```
if instanceof(option.next_scenes, Option) or len(option.next_scenes) == 1:
   return ...
weights = 
```
Always try to "early exit" instead of adding another else.